engine/gamesys/src/gamesys/scripts/script_particlefx.cpp

```diff
@@ -75,52 +75,45 @@ namespace dmGameSystem
      * @variable
      */
 
-    struct EmitterStateChangedCallbackArgs
-    {
-        EmitterStateChangedCallbackArgs(dmhash_t component_id, dmhash_t emitter_id, dmParticle::EmitterState emitter_state)
-            : m_ComponentId(component_id), m_EmitterId(emitter_id), m_EmitterState(emitter_state)
-        {}
-        dmhash_t m_ComponentId;
-        dmhash_t m_EmitterId;
-        dmParticle::EmitterState m_EmitterState;
-    };
-
-    static void EmitterStateChangedCallbackArgsCB(lua_State* L, void* user_args)
-    {
-        EmitterStateChangedCallbackArgs* args = (EmitterStateChangedCallbackArgs*)user_args;
-        dmScript::PushHash(L, args->m_ComponentId);
-        dmScript::PushHash(L, args->m_EmitterId);
-        lua_pushnumber(L, args->m_EmitterState);
-    }
-
     void EmitterStateChangedCallback(uint32_t num_awake_emitters, dmhash_t emitter_id, dmParticle::EmitterState emitter_state, void* user_data)
     {
+
         EmitterStateChangedScriptData data = *(EmitterStateChangedScriptData*)(user_data);
 
-        if (data.m_CallbackInfo)
+        if (!data.m_CallbackInfo)
```

Comment:
I prefer an early return if there's nothing to do

Reply:
The IsCallbackValid also does a null check